ansible/roles/vm_set/tasks/start_k8s_vm.yml

```diff
@@ -0,0 +1,69 @@
+- name: Create directory for Ubuntu VM disks
+  file: path={{ k8s_root_path }}/{{ vm_name }} state=directory mode=0755
+
+- name: Check destination file existence
+  stat: path={{ disk_image }}
+  register: file_stat
+
+- name: "Generate cloud-config file for {{ vm_name }}"
+  template: 
+    src: templates/cloud-config.yml.j2
+    dest: "{{ k8s_root_path }}/{{ vm_name }}/cloud-config.yml"
+  when: not file_stat.stat.exists
+
+- name: "Generate network metadata file for {{ vm_name }}"
+  template: 
+    src: templates/network-config-v2.yml.j2
+    dest: "{{ k8s_root_path }}/{{ vm_name }}/network-config-v2.yml"
+  when: not file_stat.stat.exists
+
+- name: "Generate cloud-config image for {{ vm_name }}"
+  shell: 
+    creates: "{{ k8s_root_path }}/{{ vm_name }}/{{ vm_name }}_cloud-config.img"
+    cmd: >-
+      cloud-localds 
+      --network-config={{ k8s_root_path }}/{{ vm_name }}/network-config-v2.yml
+      {{ k8s_root_path }}/{{ vm_name }}/{{ vm_name }}_cloud-config.img 
+      {{ k8s_root_path }}/{{ vm_name }}/cloud-config.yml
+  when: not file_stat.stat.exists  
+
+- name: Copy Ubuntu disk image for {{ hostname }}
+  copy: src={{ src_disk_image }} dest={{ disk_image }} remote_src=True
+  when: not file_stat.stat.exists
```

Comment:
>when: not file_stat.stat.exists [](start = 2, length = 31)

You can remove many `when` since it was handled well by corresponding modules already.

Reply:
I removed many `when` checks, just left the one so that the disk image would not get resized twice